src/platform/Zephyr/DiagnosticDataProviderImpl.cpp

```diff
@@ -52,16 +52,53 @@ const size_t kMaxHeapSize = CONFIG_SRAM_BASE_ADDRESS + KB(CONFIG_SRAM_SIZE) - PO
 
 #endif
 
+#ifdef CONFIG_SOC_SERIES_NRF54HX
+#include <hal/nrf_resetinfo.h>
+#endif
+
 namespace chip {
 namespace DeviceLayer {
 
 namespace {
 
 BootReasonType DetermineBootReason()
 {
-#ifdef CONFIG_HWINFO
+#if defined(CONFIG_SOC_SERIES_NRF54HX) || defined(CONFIG_HWINFO)
     uint32_t reason;
+#endif
+
+#ifdef CONFIG_SOC_SERIES_NRF54HX
+    reason = nrf_resetinfo_resetreas_global_get(NRF_RESETINFO);
```

Comment:
I would like to not use this API as it is HW specific.

What about to wait for this PR: https://github.com/nrfconnect/sdk-zephyr/pull/2293 as dependency. And use the API of HW_INFO instead?